### PR TITLE
[WIP] BUG: Fix name property output when name is not assigned in Series

### PR DESCRIPTION
**Pull request by charlesdong1991**
#1010

### CODE REVIEW

File: databricks/koalas/series.py [hunk 1]
```diff
@@ -323,6 +323,9 @@ def __init__(self, data=None, index=None, dtype=None, name=None, copy=False, fas
                 assert not fastpath
                 s = data
             else:
+                if name is None:
+                    self._has_no_name = True
+                self._has_no_name = False
```
Comment (ueshin) [inline databricks/koalas/series.py]: I don't think we should have this attribute. e.g.,

```py
>>> kser = ks.Series([1,2,3])
>>> kser.name = 'foo'
>>> kser
???  # maybe always None
```

How about just setting name for `self` after `IndexOpsMixin.__init__(...)`?

Comment (charlesdong1991) [reply databricks/koalas/series.py]: ahh, definitely better!!

File: databricks/koalas/series.py [hunk 2]
```diff
@@ -327,6 +327,7 @@ def __init__(self, data=None, index=None, dtype=None, name=None, copy=False, fas
                     data=data, index=index, dtype=dtype, name=name, copy=copy, fastpath=fastpath)
             kdf = DataFrame(s)
             IndexOpsMixin.__init__(self, kdf._internal.copy(scol=kdf._internal.data_scols[0]), kdf)
+            self.name = name
```
Comment (ueshin) [inline databricks/koalas/series.py]: We need to think about the case:

```py
s = ks.Series(pd.Series([1,2,3], name='foo'))
```

maybe `self.name = s.name`?

File: databricks/koalas/series.py [hunk 3]
```diff
@@ -891,7 +892,7 @@ def shape(self):
         return len(self),
 
     @property
-    def name(self) -> Union[str, Tuple[str, ...]]:
+    def name(self) -> Union[str, Tuple[str, ...], None]:
```
Comment (HyukjinKwon) [inline databricks/koalas/series.py]: I think it should be `Optional[Union[str, Tuple[str, ...]]`

Comment (HyukjinKwon) [reply databricks/koalas/series.py]: We can change the type hint for `@name.setter` below as well.

**Comment by charlesdong1991** [comment]
ah, okay, thanks for both of your comments, will take a closer look later after work! @ueshin @HyukjinKwon

**Comment by ueshin** [comment]
We can have a `None` as the name for `Series` since we have an actual Spark column in `column_scols` (previously `data_columns`), but we might need to check whether all APIs are working with it as the same as when we support `column_index`.

**Comment by itholic** [comment]
closed since it's resolved in #1712 .

**Pull request closed without merging**